alerts/system_alerts.libsonnet

```diff
@@ -187,6 +187,32 @@ local utils = import 'utils.libsonnet';
               message: 'A client certificate used to authenticate to the apiserver is expiring in less than %s.' % (utils.humanizeSeconds($._config.certExpirationCriticalSeconds)),
             },
           },
+          {
+            alert: 'KubeletPlegIntervalHigh',
+            expr: |||
+              max(kubelet_pleg_relist_interval_microseconds{quantile="0.99"} / 1000000) by (node) > %(kubeletPlegIntervalWarningSeconds)s
+            ||| % $._config,
+            'for': '10m',
+            labels: {
+              severity: 'warning',
+            },
+            annotations: {
+              message: 'The Kubelet Pod Liecycle Event Generator has a 99th percentile interval duration of {{ $value }} seconds on node {{ $labels.node }}.',
+            },
+          },
+          {
+            alert: 'KubeletPlegIntervalHigh',
+            expr: |||
+              max(kubelet_pleg_relist_interval_microseconds{quantile="0.99"} / 1000000) by (node) > %(kubeletPlegIntervalCriticalSeconds)s
+            ||| % $._config,
+            'for': '10m',
+            labels: {
+              severity: 'critical',
```

Comment:
I'm not sure anyone should be paged on this. What symtom does this incur when it happens? Pods in pending? Alerts that page should really only be based on symtoms that a user notices.

Reply:
In this particular instance newly scheduled pods onto this node were in the pending state (specifically `ContainerCreating`) and hanging.  Logs on the node itself highlighted the PLEG issue, which led me to the PLEG metric.  However, digging in further, the node itself was flapping between `Ready` and `NotReady`.

What I think happened:
- Node became unhealthy and started flapping.
- During the times when the node was considered `NotReady`, liveness probes failed and pods migrated to other nodes.
- Upon a push of a deployment update, the node just so happened to be in a `Ready` state and, since all other pods had been drained, k8s saw the node as having plenty of resources to allocate and scheduled the new pods to this node.

So I think you're right - there's no need to page here and `critical` is incorrect.

And now I'm wondering is if it is better to refine/extend the `KubeNodeNotReady` alert to detect flapping nodes instead of (or in addition to) this PLEG based alert.  Would something like `changes(kube_node_status_condition{status='true',condition='Ready'}[15m]) > 1` be appropriate?

Reply:
And FWIW, here's a couple screenshots of the node flapping data:

`kube_node_status_condition{status='true',condition='Ready'} == 0`:

![image](https://user-images.githubusercontent.com/369946/65713382-ecb68900-e04d-11e9-8892-a9183610b88e.png)

and `changes(kube_node_status_condition{status='true',condition='Ready'}[15m]) > 1`:

![image](https://user-images.githubusercontent.com/369946/65713415-ff30c280-e04d-11e9-8838-06f84e60b40e.png)

Reply:
Yes I think that's a very good alert to have too. I'd say let's add a distinct one with warning severity called `KubeNodeReadinessFlapping` or something along those lines. No strong opinion whether this or a separate PR - your call.